code/modules/awaymissions/gateway.dm

```diff
@@ -235,14 +263,56 @@ GLOBAL_DATUM(the_gateway, /obj/machinery/gateway/centerstation)
 		if (M.client)
 			M.client.move_delay = max(world.time + 5, M.client.move_delay)
 
+/obj/machinery/gateway/centeraway/unstable
+	targetid = "TIMED_MISSION"
+	var/timeinmission = 3000
+	var/collapsetime = null
+	var/list/obj/item/pinpointer/pinpointer_gateway/linked_pinpointers = list()
 
-/obj/machinery/gateway/centeraway/admin
-	desc = "A mysterious gateway built by unknown hands, this one seems more compact."
+/obj/machinery/gateway/centeraway/unstable/Initialize()
+	. = ..()
+	collapsetime = world.time + timeinmission
 
-/obj/machinery/gateway/centeraway/admin/Initialize()
+/obj/machinery/gateway/centeraway/unstable/proc/get_time()
+	var/showtime = collapsetime - world.time
+	return showtime / 10
+
+/obj/machinery/gateway/centeraway/unstable/examine(mob/user)
+	. = ..()
+	to_chat(user, "This gateway will collapse in [get_time()] seconds.")
+
+/obj/machinery/gateway/centeraway/unstable/process()
+	if(collapsetime < world.time)
+		collapse()
+	else
+		var/volume = (get_time() <= 20 ? 30 : 5)
+		playsound(loc, 'sound/items/timer.ogg', volume, FALSE)
+
+/obj/machinery/gateway/centeraway/unstable/proc/collapse()
+	for(var/obj/item/pinpointer/pinpointer_gateway/P in linked_pinpointers)
+		P.alert = FALSE
+		P.linked_gate = null
+	for(var/obj/machinery/gateway/centerstation/CG in world)
+		CG.toggleoff()
+	for(var/obj/machinery/gateway/centeraway/OG in world)
+		if(OG.z == z)
+			stationgate = null
+			targetid = null
+	for(var/mob/living/M in GLOB.mob_list)
+		if(M.z == z)
+			M.dust()
+	for(var/obj/machinery/gateway/LG in linked)
+		qdel(LG)
+	qdel(src)
```

Comment:
Are you sure it's good idea to delete it here? Doesn't it have Destroy() proc?

Reply:
Do objects with the INDESTRUCTIBLE flag work with Destroy()?